code/code/misc/offense.cc

```diff
@@ -766,332 +766,301 @@ static bool canFleeThisWay(TBeing *ch, dirTypeT dir)
-    return TRUE;
+    sendTo("You're too disoriented to flee.\n\r");
+    return true;
   }
+
   // This is used to prevent 'insta-fleeing' on mobiles
-  if (isAffected(AFF_SHOCKED) && !isPc()) {
-    sendTo("You are still recovering from something!\n\r");
-    return FALSE;
-  } if (!isPc()) {
+  if (!isPc()) {
+    if (isAffected(AFF_SHOCKED)) {
+      sendTo("You are still recovering from something!\n\r");
+      return false;
+    }
     affectedData tAff;
-
-    tAff.type     = AFFECT_COMBAT;
-    tAff.duration = 1; // Short and sweet
+    tAff.type = AFFECT_COMBAT;
+    tAff.duration = 1;
     tAff.modifier = AFF_SHOCKED;
     tAff.location = APPLY_NONE;
-
     affectTo(&tAff, -1);
   }
 
   if (isAffected(AFF_PARALYSIS)) {
     sendTo("You can't flee while paralyzed.\n\r");
-    return FALSE;
+    return false;
   }
+
   if (isAffected(AFF_STUNNED)) {
     sendTo("You can't flee while stunned.\n\r");
-    return FALSE;
+    return false;
   }
+
   if (getPosition() <= POSITION_STUNNED)
-    return FALSE;
+    return false;
 
   if (isCombatMode(ATTACK_BERSERK)) {
-    sendTo("You are berserking!  There is no way to flee!\n\r");
-    return FALSE;
-  }
-  if (bothLegsHurt() && !isFlying()) {
-    sendTo("It is futile to try to flee without use of your legs!\n\r");
-    act("$n tries to flee, but $s legs just won't move!", TRUE, this, NULL, NULL, TO_ROOM);
-    return FALSE;
-  }
-  if (eitherLegHurt() && ::number(0,2) && !isFlying()) {
-    sendTo("Your injured leg makes the attempt fail!\n\r");
-    act("$n tries to flee, but $s legs just won't move!", 
-         TRUE, this, NULL, NULL, TO_ROOM);
-    return FALSE;
+    sendTo("Flee? FLEE!? Your rage is too overwhelming!!\n\r");
+    return false;
   }
-  if (!isHumanoid() && eitherArmHurt() && ::number(0,2) && !isFlying()) {
-    // non-humanoid arms are considered legs
-    sendTo("Your injured leg makes the attempt fail!\n\r");
-    act("$n tries to flee, but $s legs just won't move!", 
-         TRUE, this, NULL, NULL, TO_ROOM);
-    return FALSE;
+
+  if (!isFlying()) {
+    if (bothLegsHurt()) {
+      sendTo("You try to flee, but can't due to the condition of your legs!\n\r");
+      act("$n tries to flee, but $s legs just won't move!", true, this, nullptr, nullptr, TO_ROOM);
+      return false;
+    }
+    if ((eitherLegHurt() || (!isHumanoid() && eitherArmHurt())) && ::number(0, 2)) {
+      sendTo("You try to flee, but can't due to the condition of your legs!\n\r");
+      act("$n tries to flee, but $s leg just won't move!", true, this, nullptr, nullptr, TO_ROOM);
+      return false;
+    }
   }
+
   if (isAffected(AFF_CHARM) && master && sameRoom(*master)) {
     if (!::number(0, 5))
-      act("$n bursts into tears.", TRUE, this, 0, 0, TO_ROOM);
-    act("You burst into tears at the thought of leaving $N.", 
-            FALSE, this, 0, master, TO_CHAR);
-    return FALSE;
+      act("$n bursts into tears at the thought of fleeing without $N!", true, this, nullptr, master,
```

Comment:
true, this, nullptr, master, TO_ROOM - no idea what these are, except maybe for master and TO_ROOM.
I'd look for ways to create a helper around act() that reduces the number of things passed.

Reply:
So after spending a little time thinking on this one, I'm not sure it's really within the scope of this PR. `act` is used hundreds (if not thousands) of times throughout the codebase in this way, so improving its usage here would only be addressing a small fraction of the problem.

It honestly is a very kludgy function - so much so that it has [its own page in the repo wiki](https://github.com/sneezymud/sneezymud/wiki/act()-format-guide) explaining how to use it. It needs a complete update at some point, instead of just a band-aid fix, IMO. If it were re-written to accept a variant in a couple places it could remove the need for a helper function at all.

If you really do want it addressed to make this PR mergeable any suggestions for how you'd go about writing the helper function would be welcome, as I'm struggling to come up with an idea that doesn't turn into just rewriting the function itself.

Reply:
No, I'm definitely not blocking the PR behind that.

Reply:
Just added https://github.com/sneezymud/sneezymud/issues/397 to track this. It's definitely a massive project.